app/models/miq_report/search.rb

```diff
@@ -104,7 +104,9 @@ def paged_view_search(options = {})
 
     if options[:parent]
       targets = get_parent_targets(options)
-      if targets.empty?
+      # filter out archived VMs
+      targets = targets.select { |target| !Vm.find_by(:id => target).try(:archived?) }
```

Comment:
What @gtanzillo says, you need to pass the correct sql conditions (!archived) from above, this code should not mention any specific classes. I assume that adding of that condition should depend on your position in the tree?

E.g. when you are on all VMs or archived, you should see archived. When you click on provider, you should not see them?